routemaster/models/queue.rb

```diff
@@ -1,130 +1,192 @@
 require 'routemaster/models'
-require 'routemaster/models/message'
-require 'routemaster/mixins/log'
 require 'routemaster/mixins/redis'
-require 'routemaster/services/codec'
+require 'routemaster/mixins/log'
+require 'routemaster/mixins/log_exception'
+require 'routemaster/models/job'
 
 module Routemaster
   module Models
-    # Abstraction for a queue.
-    # Takes a Subscriber and allows to push Messages in and pull them out in
-    # order.
+    # A collection of `Job`s, scheduled or queued, with deduplication and safe
+    # mutation (push/pop).
     class Queue
-      include Mixins::Log
       include Mixins::Redis
+      include Mixins::Log
+      include Mixins::LogException
+
+      attr_reader :name
+
+      class Retry < StandardError
+        attr_reader :delay
 
-      attr_reader :subscriber
+        DEFAULT_DELAY = 10_000 # milliseconds
```

Comment:
Well, no. It's not a maximum — and in fact the delay can go up to 96,000.
But this is ugly though; I'll remove the default as a delay explicitely gets passed in all cases.